tests/sync/test_git.py

```diff
@@ -154,6 +156,69 @@ def test_repo_update_handle_cases(
     assert mocker.call(["symbolic-ref", "--short", "HEAD"]) not in mocka.mock_calls
 
 
+@pytest.mark.parametrize(
+    "has_untracked_files,needs_stash,has_remote_changes",
+    [
+        [True, True, True],
+        [True, True, False],
+        [True, False, True],
+        [True, False, False],
+        [False, True, True],
+        [False, True, False],
+        [False, False, True],
+        [False, False, False],
```

Comment:
Behold, the grid!

![image](https://user-images.githubusercontent.com/26336/192177541-95624639-7088-4971-bd66-c7a284fa89cf.png)